addons/nuxeo-template-rendering/nuxeo-template-rendering-core/src/test/java/org/nuxeo/template/TestTemplateRenderingService.java

```diff
@@ -0,0 +1,273 @@
+@Features(CoreFeature.class)
+@Deploy("org.nuxeo.ecm.platform.query.api")
+@Deploy("org.nuxeo.ecm.platform.convert")
+@Deploy("org.nuxeo.ecm.actions")
+@Deploy("org.nuxeo.ecm.platform.rendition.api")
+@Deploy("org.nuxeo.ecm.platform.rendition.core")
+@Deploy("org.nuxeo.ecm.platform.rendition.publisher")
+@Deploy("org.nuxeo.ecm.automation.core")
+@Deploy("org.nuxeo.ecm.platform.versioning.api")
+@Deploy("org.nuxeo.ecm.platform.versioning")
+@Deploy("org.nuxeo.ecm.relations")
+@Deploy("org.nuxeo.ecm.relations.jena")
+@Deploy("org.nuxeo.template.manager")
+public class TestTemplateRenderingService {
+
+    protected static final String TEMPLATE_NAME = "mytestTemplate";
+
+    protected static final String WEBVIEW_RENDITION = "webView";
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    TemplateProcessorService tps;
+
+    private DocumentModel renditionContainer;
+
+    @Before
+    public void setup() {
+        renditionContainer = session.createDocumentModel("/", "renditions", "Folder");
+        renditionContainer = session.createDocument(renditionContainer);
+    }
+
+    @Test
+    public void whenTemplateWithoutDynamicPart_shouldRenderBlobAsIt() throws IOException {
+        TemplateSourceDocument templateSrc = createTemplateSourceDoc("Hello world !", WEBVIEW_RENDITION);
+        TemplateBasedDocument templateBase = createTemplateBasedDoc(templateSrc.getAdaptedDoc());
+
+        Blob result = templateBase.renderWithTemplate(TEMPLATE_NAME);
+        assertNotNull(result);
+        assertEquals("Hello world !", result.getString());
+    }
+
+    @Test
+    public void whenTemplateWithUsename_shouldRenderBlobAsIt() throws IOException {
+        TemplateSourceDocument templateSrc = createTemplateSourceDoc("Hello ${username} !", WEBVIEW_RENDITION);
+        TemplateBasedDocument templateBase = createTemplateBasedDoc(templateSrc.getAdaptedDoc());
+
+        Blob result = templateBase.renderWithTemplate(TEMPLATE_NAME);
+        assertNotNull(result);
+        assertEquals("Hello Administrator !", result.getString());
+    }
+
+    @Test
+    public void whenTemplateWithTemplateDocTitle_shouldRenderBlobAsIt() throws IOException {
+        TemplateSourceDocument templateSrc = createTemplateSourceDoc("Hello from ${doc['dc:title']} !", WEBVIEW_RENDITION);
+        TemplateBasedDocument templateBase = createTemplateBasedDoc(templateSrc.getAdaptedDoc());
+
+        Blob result = templateBase.renderWithTemplate(TEMPLATE_NAME);
+        assertNotNull(result);
+        assertEquals("Hello from MyTemplateBase !", result.getString());
+    }
+
+    @Test
+    public void whenTemplateWithGivenInputTemplate_String_shouldRenderBlobAsIt() throws IOException {
+        TemplateSourceDocument templateSrc = createTemplateSourceDoc("We are introducing ${myStringInCtxt} !", WEBVIEW_RENDITION);
+        TemplateBasedDocument templateBase = createTemplateBasedDoc(templateSrc.getAdaptedDoc());
+        List<TemplateInput> params = Arrays.asList(
```

Comment:
Could you use `List#of` or `Collections#singletonList`?

Reply:
done